### ISSUE TITLE
Large Ferox not extracting correctly

### ISSUE THREAD

**Issue by purplewargreymon** [cadon/ARKStatsExtractor#1300]
Large Ferox, as they are technically a separate dino from small version, need to be stat calculated as a separate entity. The issue I actually have here regarding doing so is large ferox have a plethora of issues revolving around their stat extractions when tamed/bred/imprinted. I've found a number of ways to minimize this, however melee damage no matter what ways around I find, won't extract properly. 

What I have found regarding this: 

- Post Tame
Large ferox stats will not update immediately to their tamed stats, and also do not get extra levels from taming (the 75 bonus levels go straight into wasted points). To extract them you basically have to extract it as if it was wild, then swap the button to tamed and add it. This shows the melee to be higher than it shows in game (until leveled)

- When Leveled
Large ferox stats will update to a post tame state, but melee registers in the game as lower than smart breeder registers it. Example being I had a 252.4 Dmg Ferox, and it extracts as being 255.2

- When Raised
Large ferox stats show imprinted stats when small, except they don't update melee or speed yet. When enlarged they update the speed but not the melee. Finally, when re-rendered it updates the melee. However, the melee does not extract properly. It shows in game lower than smart breeder thinks melee with 100% imprint *should* be. Example of this is I had a 249.2 Dmg Ferox show in smartbreeder as 269.3

Willing to provide further examples, screenshots and more. Whatevers needed information wise to see extractions improve and improve! ^_^

**Comment by cadon** [comment]
Hi, the large Ferox is a complicated species, mainly because after the transformation from the small state ARK doesn't display the stat values correctly and thus Smart Breeding cannot extract the levels. If you want to extract a large Ferox, make sure the values are displayed correctly, the most reliable way is to put it in a cryopod and release it before exporting. Leaving render distance sometimes works as well.
IIRC when transforming from small to large, the Ferox will move all wild levels of `Speed` to `Melee Damage`, but the according increase in the stat value of melee damage is only adjusted after a recalculation of the stats, e.g. cryo/rerender. I'm not sure how the game knows how many levels were moved from speed to melee damage, and if after a transformation from large to small the small Ferox is like before the transformation from small to large.
I don't know how the imprinting bonus is handled by the Ferox when transforming, it could be that it is lost and set to 100%.
There was also an issue that after each transformation the creature was spawned anew with a new creature id, so the last id is lost after each transformation and Smart Breeding cannot link transformed Ferox and always treats them as new creatures.

In conclusion, only import Ferox after their values have been recalculated, preferrably by using a cryopod (a soulball works as well, if you have the mod DinoStorage 2).

**Comment by purplewargreymon** [comment]
Aye largely thats what I've been doing but even then the melee isnt extracted well. Becomes a guessing game as to what the melee points actually are. Given the difficulty the game itself has with this i cant imagine this is an easy fix but figured I'd put it here anyway.

As for stats going into melee from the speed stat, i believe this were true on release, but since the change that shows the stats in parenthesis in the small ones inventory i've found this isn't the case.

What I have found is the stats don't correlate from small to large much if any at all. they're treated as two separate tames (IE I get a small ferox with 44 pts in hp, but transforms to a large ferox with 31 pts hp.) I've also found the +75 tame levels from a perfect 150 tame, while distributing randomly on the small ferox, all go straight into speed instead on the large ferox. (No points into melee anymore)

There is also a glitch where if you feed a wild ferox ele but it doesn't enlarge, an invisible large one spawns and runs around. when it shrinks, you now have 2 ferox exact same level. Further information on this dupe, is upon taming both, the small ferox gets random 75 levels allocated differently per ferox, but the large versions of both of them are identical, with no changes to their stats. This further leads me to believe the 75 levels now go into speed. Hopefully that helps if anything.

**Comment by cadon** [comment]
All extra levels in speed sounds wrong. I believe (but haven't tested), they should go into melee damage (that would be expected from a game design view to give the creature more strength). ARK is known for not updating the displayed stat values, e.g. after imprinting. If you have time to test this further I suggest to use the [Admin Blink Rifle](https://ark.wiki.gg/wiki/Admin_Blink_Rifle), e.g. in singleplayer, you can spawn it in with the console command `cheat giveitem "Blueprint'/Game/Extinction/Weapon_AdminBlinkRifle/PrimalItem_WeaponAdminBlinkRifle.PrimalItem_WeaponAdminBlinkRifle'" 1 0 0`. Using the right mouse button to switch to the green inspect mode will give you the stat level and stat value information of a creature.

**Comment by purplewargreymon** [comment]
Will test this tonight when I'm home and relay the info here thanks! appreciate the responses!
